.travis/linux_tengine_runner.sh

```diff
@@ -0,0 +1,203 @@
+#!/usr/bin/env bash
+
+set -ex
+
+export_or_prefix() {
+    export OPENRESTY_PREFIX="/usr/local/openresty-debug"
+}
+
+create_lua_deps() {
+    sudo luarocks make --lua-dir=${OPENRESTY_PREFIX}/luajit rockspec/apisix-dev-1.0-0.rockspec --tree=deps --only-deps --local
+    sudo luarocks install --lua-dir=${OPENRESTY_PREFIX}/luajit lua-resty-libr3 --tree=deps --local
+    echo "Create lua deps cache"
+    sudo rm -rf build-cache/deps
+    sudo cp -r deps build-cache/
+    sudo cp rockspec/apisix-dev-1.0-0.rockspec build-cache/
+}
+
+before_install() {
+    sudo cpanm --notest Test::Nginx >build.log 2>&1 || (cat build.log && exit 1)
+}
+
+tengine_install() {
+    if [ -d "build-cache${OPENRESTY_PREFIX}" ]; then
+        # sudo rm -rf build-cache${OPENRESTY_PREFIX}
+        sudo mkdir -p ${OPENRESTY_PREFIX}
+        sudo cp -r build-cache${OPENRESTY_PREFIX}/* ${OPENRESTY_PREFIX}/
+        ls -l ${OPENRESTY_PREFIX}/
+        ls -l ${OPENRESTY_PREFIX}/bin
+        return
+    fi
+
+    wget https://openresty.org/download/openresty-1.15.8.2.tar.gz
+    tar zxf openresty-1.15.8.2.tar.gz
+    wget https://codeload.github.com/alibaba/tengine/tar.gz/2.3.2
+    tar zxf 2.3.2
+    wget https://codeload.github.com/openresty/luajit2/tar.gz/v2.1-20190912
+    tar zxf v2.1-20190912
+    wget https://codeload.github.com/simplresty/ngx_devel_kit/tar.gz/v0.3.1
+    tar zxf v0.3.1
+
+    rm -rf openresty-1.15.8.2/bundle/nginx-1.15.8
+    mv tengine-2.3.2 openresty-1.15.8.2/bundle/
+
+    rm -rf openresty-1.15.8.2/bundle/LuaJIT-2.1-20190507
+    mv luajit2-2.1-20190912 openresty-1.15.8.2/bundle/
+
+    rm -rf openresty-1.15.8.2/bundle/ngx_devel_kit-0.3.1rc1
+    mv ngx_devel_kit-0.3.1 openresty-1.15.8.2/bundle/
+
```

Comment:
A little radical, but I like it.

Reply:
https://github.com/iresty/apisix/issues/677